services/openvpn/service/manager.go

```diff
@@ -103,12 +102,16 @@ func (m *Manager) Serve(providerID identity.Identity) (err error) {
 		IP:   net.ParseIP(m.serviceOptions.Subnet),
 		Mask: net.IPMask(net.ParseIP(m.serviceOptions.Netmask).To4()),
 	}
-	err = m.natService.Add(nat.RuleForwarding{
-		SourceSubnet: m.vpnNetwork.String(),
-		TargetIP:     m.outboundIP,
-	})
-	if err != nil {
-		return errors.Wrap(err, "failed to add NAT forwarding rule")
+
+	var dnsOK bool
+	var dnsIP net.IP
+	var dnsPort = 11153
+	m.dnsProxy = dns.NewProxy("", dnsPort)
```

Comment:
Would it be possible to run one DNS proxy for both OpenVPN and Wireguard?

Reply:
Yes, but it's more complicated if we want to keep services isolated. If the node starts up, it may be a consumer node, so we don't want to always start a DNS proxy. Just for provider nodes.